binary-option/program/src/processor.rs

```diff
@@ -0,0 +1,725 @@
+        match instruction {
+            BinaryOptionInstruction::InitializeBinaryOption(args) => {
+                msg!("Instruction: InitializeBinaryOption");
+                process_initialize_binary_option(program_id, accounts, args.decimals)
+            }
+            BinaryOptionInstruction::Trade(args) => {
+                msg!("Instruction: Trade");
+                process_trade(
+                    program_id,
+                    accounts,
+                    args.size,
+                    args.buy_price,
+                    args.sell_price,
+                )
+            }
+            BinaryOptionInstruction::Settle => {
+                msg!("Instruction: Settle");
+                process_settle(program_id, accounts)
+            }
+            BinaryOptionInstruction::Collect => {
+                msg!("Instruction: Collect");
+                process_collect(program_id, accounts)
+            }
+        }
+    }
+}
+
+pub fn process_initialize_binary_option(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    decimals: u8,
+) -> ProgramResult {
+    msg!("InitializeBinaryOption");
+    let account_info_iter = &mut accounts.iter();
+    let binary_option_account_info = next_account_info(account_info_iter)?;
+    let escrow_mint_info = next_account_info(account_info_iter)?;
+    let escrow_account_info = next_account_info(account_info_iter)?;
+    let long_token_mint_info = next_account_info(account_info_iter)?;
+    let short_token_mint_info = next_account_info(account_info_iter)?;
+    let mint_authority_info = next_account_info(account_info_iter)?;
+    let update_authority_info = next_account_info(account_info_iter)?;
+    let token_program_info = next_account_info(account_info_iter)?;
+    let system_account_info = next_account_info(account_info_iter)?;
+    let rent_info = next_account_info(account_info_iter)?;
+
+    create_new_account(
+        &mint_authority_info,
+        &long_token_mint_info,
+        Mint::LEN,
+        &token_program_info,
+        &rent_info,
+    )?;
+    create_new_account(
+        &mint_authority_info,
+        &short_token_mint_info,
+        Mint::LEN,
+        &token_program_info,
+        &rent_info,
+    )?;
+    create_new_account(
+        &update_authority_info,
+        &escrow_account_info,
+        Account::LEN,
+        &token_program_info,
+        &rent_info,
+    )?;
+    spl_mint_initialize(
+        &token_program_info,
+        &long_token_mint_info,
+        &mint_authority_info,
+        &mint_authority_info,
+        &rent_info,
+        0,
+    )?;
+    spl_mint_initialize(
+        &token_program_info,
+        &short_token_mint_info,
+        &mint_authority_info,
+        &mint_authority_info,
+        &rent_info,
+        0,
+    )?;
+    spl_initialize(
+        &token_program_info,
+        &escrow_account_info,
+        &escrow_mint_info,
+        &update_authority_info,
```

Comment:
I'm pretty sure it can't because I call `SetAuthority` for all of the sensitive accounts in the `Initialize` instruction. The update authority just creates everything and then immediately relinquishes authority

Reply:
Ah oops sorry, you can disregard this, I got confused from all of the accounts that are passed in and then not really used, and then forgot to delete this comment